figures/mapping-histology-labels.Rmd

```diff
@@ -72,89 +63,73 @@ output_dir <- "palettes"
 Which variables are we keeping for this table? 
 
 ```{r}
-histology_variables <- 
-  c("integrated_diagnosis", 
-    "Notes", 
-    "harmonized_diagnosis",
-    "broad_histology", 
-    "short_histology")
+histology_variables <- "cancer_group"
 ```
 
-Let's read in the current release's `pbta-histologies.tsv` file. 
+Let's read in the current release's `histologies.tsv` file. 
 
 ```{r}
 metadata <-
-  readr::read_tsv(file.path(input_dir, "pbta-histologies.tsv"), guess_max = 10000)
+  readr::read_tsv(file.path(input_dir, "histologies.tsv"), guess_max = 100000)
 ```
 
 Now we'll select histology variables we mentioned above and so capitalization differences don't get in the way with this process, we will change everything to lower case for now. 
 
 ```{r}
 working_metadata <- metadata %>% 
-  dplyr::select(Kids_First_Biospecimen_ID, sample_type, histology_variables) %>% 
-  dplyr::mutate(broad_histology_lower = tolower(broad_histology))
+  dplyr::select(Kids_First_Biospecimen_ID, sample_type, histology_variables) 
 ```
 
-# Take a look at how many biospecimens per `broad_histology` group
+# Take a look at how many biospecimens per `cancer_group` group
 
-Let's summarize `broad_histology`. 
-Because the `Normal` samples don't have histologies, we'll look at just the `Tumor` samples at for this summary. 
+Let's summarize `cancer_group`. 
+Because the `Normal` samples don't have `cancer_group`, we'll look at just the `Tumor` samples at for this summary. 
 
 ```{r}
-broad_summary <- working_metadata %>% 
+cancer_group_summary <- working_metadata %>% 
   dplyr::filter(sample_type == "Tumor") %>%
-  dplyr::count(broad_histology_lower) %>% 
+  dplyr::count(cancer_group) %>% 
+  dplyr::arrange(n) 
+```
+
+For plotting purposes, we only want to plot cancer groups with n>=5, hence when generating the file, we only need colors for those samples
+```{r}
+cancer_group_summary <- working_metadata %>% 
+  dplyr::filter(sample_type == "Tumor") %>%
+  dplyr::count(cancer_group) %>% 
   dplyr::arrange(n) 
 ```
 
 Let's print out the summary. 
 
 ```{r}
-broad_summary %>% 
+cancer_group_summary  %>% 
   knitr::kable()
 ```
 
 There's handful of very small groups (many are n = 2). 
 
 ## Declare new equivalent groups
-
-These groups we'll combine into a non-CNS/other tumor group.
-
-```{r}
-other_tumor <- c("lymphoma", "melanocytic tumor", "other tumor", "metastatic tumors", "non-cns tumor")
-```
-
-These groups we'll combine as a benign.
-
-```{r}
-benign <- c("benign tumor", "non-tumor")
-```
-
-Add in the `Other astrocytic tumor` in with the LGAT group. 
-
-```{r}
-lgat <- c("other astrocytic tumor", "low-grade astrocytic tumor")
-```
```

Comment:
@kgaonkar6, I looked into the histologies file and felt like those lines no longer make sense - could you please confirm it's ok to delete these lines?